### PR TITLE
Fix oba msids

### PR DESCRIPTION
**Pull request by matthewdahmer**
@taldcroft 
@jeanconn 

This update includes a fix for the 4OAVOBAT telescope msid which continues to show the narrow range version of this msid where it should be reporting the wide range version after the associated patch was uplinked. This update also includes a minor fix for the DP_OBA_AVE derived parameter.

### CODE REVIEW

**Comment by taldcroft** [comment]
Try to do this without repeating code.  Hint, with good surrounding comments you can hijack the `msid_nums` dict a bit:
```
'b': '30 31 33 34 35 36 37 38 39 40 41 44 45 46 49 50 51 52 53 54 4OAVOBAT'.split()
```
Then you just need to handle the oddball in the loop.

**Comment by matthewdahmer** [comment]
Ok that works too. I wanted to avoid messing with the logic in the loop, but the result does look cleaner.

File: Ska/engarchive/converters.py [hunk 1]
```diff
@@ -366,8 +366,12 @@ def obc4eng(dat):
...
         if np.any(mask):
             for msid_num in msid_nums[patch]:
-                msid = 'OOBTHR' + msid_num
-                msid_wide = msid + '_WIDE'
+                if '4OAVOBAT' in msid_num:
+                    msid = '4OAVOBAT'
+                    msid_wide = '4OAVOBAT_WIDE'
+                else:
+                    msid = 'OOBTHR' + msid_num
+                    msid_wide = msid + '_WIDE'
```
Comment (taldcroft) [inline Ska/engarchive/converters.py]: You can pull this line out of the `else:` and remove line 371 above.

**Comment by matthewdahmer** [comment]
True, that eliminates another line, new commit to follow momentarily.

**Comment by taldcroft** [comment]
@matthewdahmer - I rebased this on the `state-bins` branch prior to regenerating data.

**Comment by matthewdahmer** [comment]
@taldcroft  - Thank You!
I take it I will not need to manually update this data then correct?

**Comment by taldcroft** [comment]
@matthewdahmer - there is a slight problem with the 4OAVOBAT patch, which is that it occurs in the TEL2ENG data content, not OBC4ENG like the others.  So I would suggest that you back out the changes in `def obc4eng(..)` and make a new function that is fairly similar but is named `def tel2eng()` and fixes 4OAVOBAT.  Does 4OAVHRMT need to have the same patch, or was that not changed on board?  In the `def tel2eng()` function you probably don't need all the a/b logic since there was just one patch time.  Sorry for not noticing this earlier.

**Comment by taldcroft** [comment]
BTW, because I have rebased this onto the other branch, you need to get into your git repo and do something like:
```
git branch -m fix_oba_msids fix_oba_msids_bak
git fetch origin
git checkout fix_oba_msids
<edit as normal>
git push origin fix_oba_msids
```
Without moving the original branch out of the way there will be a merge conflict later when you try to push back up.

**Comment by matthewdahmer** [comment]
That should do it. I should have noticed that I was using the wrong data content directory. 4OAVHRMT is not affected as the constituent HRMA zones were not switched to wide range. Please let me know if you have any comments or suggestions.

**Comment by taldcroft** [comment]
OK thanks, looks good.

**Comment by taldcroft** [comment]
Relevant commits included (in squashed form) in #127.  Closing.

**Pull request closed without merging**